### PR TITLE
Update annotation form

### PR DESCRIPTION
**Pull request by jorg-vr**
This pull request restructures the annotation form in such a way that the saved annotations options don't take up to much extra space.

New:
image
Old:
image

New (save annotation):
image
Old (save annotation):
image

New (mobile):
image
Old (mobile):
image

Part of #4564

### CODE REVIEW

**Review by bmesuere — CHANGES_REQUESTED** [review]
Feedback from running this on metra:
- [x] the hover button text shows ask question instead of add comment (the global button does show add comment)
- [x] if there are no saved annotations, the area to the right of the textarea is left blank. I would add an empty state there or show the textarea full width.
- [x] if you check "save annotation", the title form is shown to the right of the text area. This somewhat breaks the flow. Maybe placing it inline after the checkbox + save annotation text would make more sense?
- [ ] ~~when save annotation is checked, it would make sense to change the text on the submit button to also say something that hints that the annotation will be saved.~~
- [x] the help text under the "search for annotations" combobox is too long (at least in Dutch)
- [x] in the search annotations input, the list of suggestions is not the same width as the text input
- [ ] in the search annotations input, the list of suggestions "flickers" when searching
- [x] when picking a suggestion, the "save annotation" checkbox is disabled, but it is also shown as checked which might be confusion. Maybe replace it with a link to the page where you can edit that annotation?
- [x] when loading a saved annotation and then changing the text, a "not equals" sign is shown in the search annotation input. The icon is not aligned well (shown somewhere in the middle of the input field)

**Comment by jorg-vr** [comment]
> when picking a suggestion, the "save annotation" checkbox is disabled, but it is also shown as checked which might be confusion. Maybe replace it with a link to the page where you can edit that annotation?

This page does not exist yet (It is a modal currently)
I will just hide the checkbox for now and reintroduce that link when that page is created.
Added this note in https://github.com/dodona-edu/dodona/issues/4564

**Comment by jorg-vr** [comment]
> when save annotation is checked, it would make sense to change the text on the submit button to also say something that hints that the annotation will be saved.

I am not the biggest fan of this. I find it hard to come up with better wording for the button. A longer text is also makes it harder to properly layout the buttons (as they are on the same line as the checkbox currently)

The checkbox is also right next to the button, so the information is already provided on almost the same place

**Comment by jorg-vr** [comment]
>  the help text under the "search for annotations" combobox is too long (at least in Dutch)

I removed `Saved annotations are automatically filtered by course and exercise. `
This will change in version 2 anyway

**Review by niknetniko — APPROVED** [review]

**Review by bmesuere — CHANGES_REQUESTED** [review]
- [x] The wording of "sla opmerking op om te hergebruiken" is quite verbose. Maybe make it "Opmerking opslaan" (maybe with an info icon with tooltip?). This would also help with my previous comment that it wasn't really clear that the comment would also be save when clicking the submit button
-  [ ] The "Titel:" label is a bit strange. Maybe [floating labels](https://getbootstrap.com/docs/5.3/forms/floating-labels/) are an option here?
- [x] Upon clicking "sla opmerking op", the titel input should get the focus
- [ ] On smaller screens, the form layout isn't optimal:
  image

**Comment by jorg-vr** [comment]
> The "Titel:" label is a bit strange. Maybe [floating labels](https://getbootstrap.com/docs/5.3/forms/floating-labels/) are an option here?

The floating labels don't work for our inputs, as we have a lot less vertical padding compared to the bootstrap default. Thus their is no vertical space for a label inside the input, without making the input look comically large compared to others.

I did make some custom styling based on the bootstrap css to put the label inside the input before the typing area 
image

File: app/assets/stylesheets/components/code_listing.css.scss [hunk 1]
```diff
@@ -478,35 +478,44 @@
...
-    }
 
-    .annotation-submission-input {
-      border-radius: 0;
-      resize: vertical;
+      label {
+        color: rgba(var(--d-on-surface-rgb), 0.75);
+        transform: scale(0.85) translateY(-0.4rem);
+        position: absolute;
+        top: 0;
```
Comment (bmesuere) [inline app/assets/stylesheets/components/code_listing.css.scss:503]:
```suggestion
        top: 1px;
```

In my Chrome, the label was not aligned with the textbox text.

File: app/assets/javascripts/components/annotations/annotation_form.ts [hunk 2]
```diff
@@ -176,95 +188,120 @@ export class AnnotationForm extends watchMixin(ShadowlessLitElement) {
         this.inputRef.value.focus();
     }
 
+    updated(changedProperties: Map<string, any>): void {
+        // Focus the newly shown title input if the user wants to save the annotation.
+        if (changedProperties.has("saveAnnotation") && this.saveAnnotation) {
+            this.titleRef.value.focus();
```
Comment (bmesuere) [inline app/assets/javascripts/components/annotations/annotation_form.ts:194]: I would also call `select()` here: https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/select

**Review by bmesuere — APPROVED** [review]
There's a JS error on mestra when signed in as staff. Otherwise, looks good 👍

**Pull request merged**